lib/src/core/comms/secure_channel.rs

```diff
@@ -35,6 +35,9 @@ pub enum Role {
     Server,
 }
 
+// Number of stored remote keys.
+const STORED_REMOTE_KEYS: usize = 5;
```

Comment:
Is it really possible for more than two tokens to be live at the same time? From reading the standard it seems like having three or more tokens overlapping would be impossible.

Reply:
It might be easier to just manually track `last_token_id` and `last_remote_key`, or something like that.

Reply:
Not sure if that is possible but the spec mentions clients should accept messages secured by an expired SecurityToken for up to 25 % of the token lifetime. So not sure if it can happen that rapid renews (for whatever reason) could occur within that time.

I looked at how other packages solved this and they generally they seem to use a map where the token ID is the key and the value are the keys.

Initially I started out with a map as well, but if you want to keep that from growing endlessly you also need to rely on the token ID being ordered. Which is again not part of the spec. It's only required for them to be unique.

But I do want to make it robust and agree that currently it isn't. Have an other idea, will have another look tomorrow...

Reply:
We are the ones that renew tokens, I believe, so we do actually have complete control over what overlap is allowed. Since we only ever renew once 75% of the token lifetime has expired, there can never be more than two active tokens at a time.